### DISCUSSION TITLE
how to use other generator than Ninja??

### DISCUSSION THREAD

**Discussion by cloudzooming** [microsoft/vcpkg#39145] [General]
I'm running windows 10 with 32 GB ram. when I try to install poco library, it exhausts all ram and makes the system freeze. But if I specify legacy baseline for example: 7249fa4d64ead1428249895810146134be8eb881 in vcpkg.json when manifest is enabled,  it builds instantly.  the legacy baseline seems not using Ninja as the generator. 

My question, is how to use "Visual Studio 17 2022" to replace "Ninja" for the ports building process??

**Comment by dg0yt** [comment]
First, you should try to limit concurrency, by setting environment variable `VCPKG_MAX_CONCURRENCY`.
https://learn.microsoft.com/en-us/vcpkg/users/config-environment#vcpkg_max_concurrency

**Reply by cloudzooming** [reply]
> First, you should try to limit concurrency, by setting environment variable `VCPKG_MAX_CONCURRENCY`. https://learn.microsoft.com/en-us/vcpkg/users/config-environment#vcpkg_max_concurrency

thanks. but it still does not work. 

Finally solved it by overriding poco version to 1.12. 

 It seems the latest version of poco problem, which causes sticking at compiling and linking odbc TypeInfo.cpp
